src/Expr.h

```diff
@@ -109,6 +109,7 @@ enum ExprTag : int {
     EXPR_FROM_ANY_VEC_COERCE,
     EXPR_ANY_INDEX,
     EXPR_SCRIPT_OPT_BUILTIN,
+    EXPR_TYPE,
```

Comment:
per the comment above this block, these are meant for internal expressions created during AST transformation, so the tag should come earlier